apollo-api/src/commonMain/kotlin/com/apollographql/apollo3/api/internal/Mutator.kt

```diff
@@ -0,0 +1,15 @@
+package com.apollographql.apollo3.api.internal
+
+/**
+ * Represents an operation that accepts a single input argument, mutate it and returns no result
+ *
+ * @param <T> the type of the input to the operation
+</T> */
+interface Mutator<T> {
```

Comment:
Referenced from the [generated toBuilder methods](https://github.com/apollographql/apollo-kotlin/pull/4361/files#diff-a5874eb3e26a618a146c7a7ceb6337778e2a398d8f0f9fecda79737ca26712f8R152) and ported from [Mutator.java](https://github.com/apollographql/apollo-kotlin/blob/66cd44629e374b667d16ee831f5a4d42cc38c50b/apollo-api/src/jvmMain/java/com/apollographql/apollo/api/internal/Mutator.java).

Reply:
This doesn't seem to be used in any test. Can you modify or add a CodegenTest for this?

Reply:
I actually didn't find any tests for it in Apollo Android, but I will check again. It seems to be not very well documented how it's supposed to work.

Reply:
If there are no tests for this, I'd just leave it out. It's always easier to add features later than removing something that's illy specified.

Reply:
From the perspective of someone migrating from Apollo 2 to 3, if I were using `toBuilder` I would expect it to still be there in Apollo 3, even if it wasn't very well documented.

Reply:
>  if I were using `toBuilder`

Given that it was not tested, I'm not sure how many users use it. What's more, one can use `toBuilder()` without any parameter right? 

```java
MyQuery.Data data = MyQuery.Data.builder().[...].build()

data.toBuilder()
  .name("foo") // change name
  .build()
```

Not sure where `Mutator` is required up there?

Reply:
I found the original PR: https://github.com/apollographql/apollo-kotlin/pull/680

At the time, you asked the author to add integration tests, but I didn't see any!

I guess it's your decision whether to support this functionality in Apollo 3.

Reply:
> I guess it's your decision whether to support this functionality in Apollo 3. 

I think it's fine to leave it out for now. It's always easier to add it later than remove it.

Reply:
Okay, I'll leave `toBuilder` but remove mutator support.